### ISSUE TITLE
Via API add sources/user_connections for a user

### ISSUE THREAD

**Issue by qdrddr** [goauthentik/authentik#8761]
Dear Authentik community,

**Describe your question/**
Please help me with API v2024.2.1/v2024.2.2: I want to create sources/user_connections for a user.
In Admin Interface > Directory > Federation and Social Logins, I have added an OAuth Source called "Password" (Auth0.com).

When I log in via UI and go to the User Settings > Connected Services, I can press the Connect button, successfully connect the service, and then log in via that Social login to Authentik.

I'm trying to do the same via API.

When I do curl with GET https://yyy.com/api/v3/sources/user_connections/all/ or with GET /api/v3/sources/user_connections/all/20/
I get the connected Service listed for the user. 
```
curl -X GET "https://yyy.com/api/v3/sources/user_connections/oauth/20/" \
 -H "accept: application/json"\
 -H "authorization: Bearer token1" 
```

RESPONSE IS:

```
{
  "pk": 20,
  "user": 4,
  "source": {
    "pk": "5ecb550b-5a50-5fd5-b55b-555cb5555555",
    "name": "Password",
    "slug": "password",
    "enabled": true,
    "authentication_flow": "44f444a4-444a-4c44-4444-c4ba44c44fd4",
    "enrollment_flow": "6d6cb66b-a6bc-6a66-b66f-f6666c66aabc",
    "component": "",
    "verbose_name": "source",
    "verbose_name_plural": "sources",
    "meta_model_name": "authentik_core.source",
    "policy_engine_mode": "any",
    "user_matching_mode": "identifier",
    "managed": null,
    "user_path_template": "goauthentik.io/sources/%(slug)s",
    "icon": "https://www.auth0.com/file/auth0-logo-icon.png"
  },
  "identifier": "auth0|xxxxxxxxxxxxxxx"
}
```

Now, I disconnect the service in the UI User Settings > Connected Services.

And want to repeat the same steps with API instead:

```
curl -X POST "https://yyy.com/api/v3/sources/user_connections/" \
     -H "accept: application/json" \
     -H "authorization: Bearer token1" \
     -H "content-type: application/json" \
     -d '{
  "user": 4,
  "identifier": "auth0|xxxxxxxxxxxxxxx"
}' -v 
```

### First problem: 
The curl returns "HTTP/2 405".

### Second problem:
The connection from above that I'm trying to create is also supposed to be linked with the service `"source": { "pk": "5ecb550b-5a50-5fd5-b55b-555cb5555555"}` in this case, but I do not see any documentation that points out how to link the two. 

**How, programmatically, via API, can I add to a user a Connected Service?**

Thank you! 

**Relevant info**
v2024.2.1 & v2024.2.2

**Screenshots**

**Logs**
Output of docker-compose logs or kubectl logs respectively

**Version and Deployment (please complete the following information):**

-   authentik version: 2024.2.1 & v2024.2.2
-   Deployment: helm

**Additional context**

**Comment by rissson** [comment]
Have you checked out the API browser? You'll find the relevant endpoints here: https://docs.goauthentik.io/developer-docs/api/browser#tag/sources/operation/sources_user_connections_oauth_update

You can also test them out on your instance directly if you go to authentik.domain/api/v3/

**Comment by qdrddr** [comment]
@rissson yes, I did check the [API Browser](https://docs.goauthentik.io/developer-docs/api/browser#tag/sources/operation/sources_user_connections_oauth_update), and as you can see I'm using that API endpoint, and it doesn't work. Please see in my comment [above](https://github.com/goauthentik/authentik/issues/8761#issue-2162438075) starting with:

```
curl -X POST "https://yyy.com/api/v3/sources/user_connections/" \
     -H "accept: application/json" \
     -H "authorization: Bearer token1" \
     -H "content-type: application/json" \
     -d '{
  "user": 4,
  "identifier": "auth0|xxxxxxxxxxxxxxx"
}' -v 
```
**First problem:**
The curl returns "HTTP/2 405".

**Second problem:**
The connection from above that I'm trying to create is also supposed to be linked with the service "source": { "pk": "5ecb550b-5a50-5fd5-b55b-555cb5555555"} in this case, but I do not see any documentation that points out how to link the two.

How, programmatically, via API, can I add to a user a Connected Service?

**Comment by qdrddr** [comment]
I also have tried these as a random guess:

```
curl -X POST "https://yyy.com/api/v3/sources/user_connections/oauth" \
     -H "accept: application/json" \
     -H "authorization: Bearer token1" \
     -H "content-type: application/json" \
     -d '{
  "user": 4,
  "source": "5ecb550b-5a50-5fd5-b55b-555cb5555555",
  "identifier": "auth0|xxxxxxxxxxxxxxx"
}' -v
```


```
curl -X POST "https://yyy.com/api/v3/sources/user_connections/oauth" \
     -H "accept: application/json" \
     -H "authorization: Bearer token1" \
     -H "content-type: application/json" \
     -d '{
  "user": 4,
  "source": {
    "pk": "5ecb550b-5a50-5fd5-b55b-555cb5555555"
  },
  "identifier": "auth0|xxxxxxxxxxxxxxx"
}' -v
```

And this also

```
curl -X POST "https://yyy.com/api/v3/sources/user_connections/oauth/5ecb550b-5a50-5fd5-b55b-555cb5555555" \
     -H "accept: application/json" \
     -H "authorization: Bearer token1" \
     -H "content-type: application/json" \
     -d '{
  "user": 4,
  "identifier": "auth0|xxxxxxxxxxxxxxx"
}' -v
```

**Comment by rissson** [comment]
No you're right, we actually forbid this in the API (https://github.com/goauthentik/authentik/blob/main/authentik/core/api/sources.py#L197). As for the reason why, let's ask @BeryJu

**Comment by qdrddr** [comment]
Okay, so it looks like there are two problems:

1) speaking from [this function](https://github.com/goauthentik/authentik/blob/main/authentik/core/api/sources.py#L197), looks like the schema is supposed to be

- "user"
- "source" - is actually there but just not documented
- And there is no identifier - which is incorrect, and it should be there

2) The class is read-only, and it should be writable, I guess.

**Comment by qdrddr** [comment]
@BeryJu & @rissson can we safly declare this is a bug and should be fixed?

**Comment by qdrddr** [comment]
Any updates on this?

**Comment by qdrddr** [comment]
Can someone please comment on this?

**Comment by qdrddr** [comment]
Hope this will be addressed some day...

**Comment by qdrddr** [comment]
Hope this will be addressed some day...

**Comment by rissson** [comment]
This was indeed missing for SAML and Kerberos, but it should be available for OAuth already. For instance:

```
POST /api/v3/sources/user_connections/oauth/
Content-Type: application/json
Authorization: Bearer bla

{
  "user": 4,
  "source": "5b8ef5de-097d-493e-8f61-2daa24770de7",
  "identifier": "oidc_sub"
}
```

**Comment by Lykos153** [comment]
I still can't get this to work for oauth. With this request

~~~
POST /api/v3/sources/user_connections/oauth/
Content-Type: application/json
Authorization: Bearer <mytoken>

{
  "user": 4,
  "source": "5b8ef5de-097d-493e-8f61-2daa24770de7",
  "identifier": "oidc_sub"
}
~~~

I get

~~~
  × HTTP Error 405 (Method Not Allowed): /api/v3/sources/user_connections/oauth/
~~~
